api_options.go

```diff
@@ -0,0 +1,133 @@
+package metaphor
+
+type ClientOptions func(*MetaphorClient)
+
+// WithNumResults sets the number of expected search results.
+//
+// Parameters:
+//  - numResults: The desired number of results.
+//
+// Returns: a ClientOptions function that updates the numResults field of the RequestBody struct.
```

Comment:
any advance of having it structured this way instead of having a struct of the options that can be passed to the functions?

Reply:
I've been using Go for about a year(but have been coding for almost 20 years), and coming from OOP(like ruby, php, ObjC) languages that would be my instinct as well. But in the past few months I've learned from the langchain-go community, and all the Go devs that the way to do it is using these options functions. 

I did a few searches (using metpahor ofc :)) due to your questions, it's a pattern, Functional Options Pattern in go, so this might clarify a bit... 
- https://michalzalecki.com/golang-options-pattern/
- https://derekchiang.com/posts/reusable-and-type-safe-options-for-go-apis/

Personally, it seems far more inline with go style and it's intuitive to use, but I don't have enough experience to make a list of sensible pros and cons. I've just picked it up 'cause all the GoLang devs are doing it, and I had PR's rejecedt asking to implement this pattern... 🤷